lms/djangoapps/mobile_api/friends/views.py

```diff
@@ -0,0 +1,96 @@
+"""
+Views for friends info API
+"""
+
+from rest_framework import generics, permissions, status
+from rest_framework.authentication import OAuth2Authentication, SessionAuthentication
+from rest_framework.response import Response
+from openedx.core.djangoapps.user_api.api.profile import preference_info
+from opaque_keys.edx.locations import SlashSeparatedCourseKey
+from social.apps.django_app.default.models import UserSocialAuth
+from student.models import CourseEnrollment
+import serializers
+import urllib2
+import json
+import facebook
+
+# TODO: change this to final config
+from ..settings import FB_SETTINGS
+_FACEBOOK_API_VERSION = FB_SETTINGS['_FACEBOOK_API_VERSION']
+
+
+class FriendsInCourse(generics.ListAPIView):
+    """
+    **Use Case**
+
+        API endpoint that returns all the users friends that are in the course specified.
+        Note that only friends that allow their courses to be shared will be included.
+
+    **Example request**:
+
+        GET /api/mobile/v0.5/friends/course/<course_id>
+
+        where course_id is in the form of /edX/DemoX/Demo_Course
+
+    **Response Values**
+
+        {   "friends":
+                [{
+                    "name": "test",
+                    "id": "12345",
+                    "edX_id": "678910"
+                },
+                ...
+                ]
+            }
+    """
+    authentication_classes = (OAuth2Authentication, SessionAuthentication)
+    permission_classes = (permissions.IsAuthenticated,)
+    serializer_class = serializers.FriendsInCourseSerializer
+
+    def list(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.GET, files=request.FILES)
+        if serializer.is_valid():
+            # Get all the users FB friends
+            graph = facebook.GraphAPI(serializer.object['oauth_token'])
+            url = _FACEBOOK_API_VERSION + "me/friends"
+            friends = graph.request(url)
+            data = self.get_pagination(friends)
+            # For each friend check if they are a linked edX user
+            friends_that_are_edX_users = []
+            for friend in data:
+                query_set = UserSocialAuth.objects.filter(uid=unicode(friend['id']))
+                if query_set.count() == 1: 
+                    friend['edX_id'] = query_set[0].user_id
+                    friend['edX_username'] = query_set[0].user.username
+                    friends_that_are_edX_users.append(friend)
+            # Filter by sharing preferences
+            friends_that_are_edX_users_with_sharing = [friend for friend in friends_that_are_edX_users if self.sharing_pref_true(friend)]
+            if ('course_id' in kwargs) and len(kwargs['course_id'].split('/')) == 3:
+                course_path = kwargs['course_id'].split('/')
+                course_key = SlashSeparatedCourseKey(course_path[0], course_path[1], course_path[2])
```

Comment:
We cannot assume the format of Course Ids.  In fact, the Course ID for new courses now use `+` separators instead of `/`.  :)

So, this code should instead be:
```
if ('course_id' in kwargs)
   course_key = CourseKey.from_string(kwargs['course_id'])
```